some minor fixes

lispc opened the pull request:
A collection of fixes of glitches found recently

1. fix some misleading log like "EVM Proof generation with {} rows".  No, degree is not rows.
2. replace Fp with Fr. Fp makes non sense here.
3. use strum to derive  ExecutionState::iter()
4. fix some clippy problem with latest rust nightly. (We will upgrade rust someday, so fix problems earlier is not bad. BTW any plan to upgrade to latest nightly and Rust2021?). Mainly `flat_map` and array into_iter() . See diffs for details.
5. re-enable some state circuit tests for opcodes, which are disabled due to historical reasons.

lispc commented on Makefile:
```diff
@@ -5,7 +5,7 @@ help: ## Display this help screen
 clippy: ## Run clippy checks over all workspace members
 	@cargo check --all-features
-	@cargo clippy --all-features --all-targets -- -D clippy::all
+	@cargo clippy --all-features --all-targets -- -D warnings
```
same with CI

CPerezz replied:
Looks fine to me.

lispc commented on circuit-benchmarks/build.rs:
```diff
@@ -7,7 +7,7 @@ use std::io::Write;
 fn main() {
     let degree: usize = var("DEGREE")
-        .unwrap_or_else(|_| "11".to_string())
+        .unwrap_or_else(|_| "18".to_string())
```
both state circuit and evm circuit needs degree 17 or 18 now

CPerezz commented on Makefile:
```diff
@@ -17,6 +17,9 @@ fmt: ## Check whether the code is formated correctly
 test: ## Run tests for all the workspace members
 	@cargo test --release --all --all-features --exclude integration-tests --exclude circuit-benchmarks
 
+test_doc: ## Test the docs
+	@cargo test --release --all --all-features --doc
+
```
If you add this new rule, please add it to the `test-all` rule too.

CPerezz commented on bus-mapping/src/circuit_input_builder/access.rs:
```diff
@@ -4,7 +4,7 @@ use ethers_core::utils::get_contract_address;
 
 /// State and Code Access with "keys/index" used in the access operation.
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Eq)]
```
Is `Eq` needed specifically?? If not, let's save space and compilation time and not derive traits unless they're required.

lispc replied:
make sense

CPerezz commented on bus-mapping/src/circuit_input_builder/access.rs:
```diff
@@ -26,7 +26,7 @@ pub enum AccessValue {
 
 /// State Access caused by a transaction or an execution step
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Eq)]
```
ditto.

CPerezz commented on bus-mapping/src/circuit_input_builder/access.rs:
```diff
@@ -50,12 +50,11 @@ fn get_call_result(trace: &[GethExecStep]) -> Option<Word> {
 
 /// State and Code Access set.
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Eq)]
```
dirro.

CPerezz commented on bus-mapping/src/circuit_input_builder/call.rs:
```diff
@@ -3,7 +3,7 @@ use crate::{exec_trace::OperationRef, Error};
 
 /// Type of a *CALL*/CREATE* Function.
-#[derive(Clone, Copy, Debug, PartialEq)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
```
ditto

CPerezz commented on circuit-benchmarks/src/keccak_permutation.rs:
```diff
@@ -170,7 +170,7 @@ mod tests {
 
         // Bench proof generation time
-        let proof_message = format!("EVM Proof generation with {} rows", degree);
+        let proof_message = format!("Keccak Proof generation with {} degree", degree);
```
Thanks for the catch!!

CPerezz commented on zkevm-circuits/src/evm_circuit.rs:
```diff
@@ -186,8 +186,8 @@ pub mod test {
 
-    pub(crate) fn rand_fp() -> Fp {
-        Fp::rand()
+    pub(crate) fn rand_scalar() -> Fr {
+        Fr::rand()
```
Is this fn used anywhere?? 

If not, we could remove it directly. Since is the same to do `Fr::rand()` than `rand_scalar()`

CPerezz commented on zkevm-circuits/src/evm_circuit/execution/memory.rs:
```diff
@@ -214,9 +214,7 @@ mod test {
-            // we have to disable state circit now, since the memory size used
-            // here is too large
-            enable_state_circuit_test: false,
+            enable_state_circuit_test: true,
```
We could even remove this field entirely if we're not considering to test without this anymore.

CPerezz commented on zkevm-circuits/src/util.rs:
```diff
@@ -54,7 +54,7 @@ impl<F: FieldExt> Expr<F> for i32 {
     fn expr(&self) -> Expression<F> {
         Expression::Constant(
-            F::from(self.abs() as u64)
+            F::from(self.unsigned_abs() as u64)
```
I still prefer `unsigned_abs` TBH. I know this should never be an issue. But since is the same, let's be sure we always get uints

CPerezz reviewed (commented):
Everything looks good. 

Not sure why do we need to derive `Eq` for every struct when it might not be needed.
Is there any particular reason??

For the rest, a couple of comments but so far good job!!!

lispc commented:
updated.

remove the newly added "Eq" trait.  When some day we upgrade rust nightly, we need to add a clippy allow rule then. Same for unsigned_abs.

I removed "enable_state_circuit_test: false," in memory opcode(it was disabled due to the previous state circuit design, fixed now). 
And use "enable_state_circuit_test: true," in storage opcode since mpt circuit <-> state circuit is still under development. May needed to be disabled & enabled from time to time.

CPerezz commented:
> remove the newly added "Eq" trait. When some day we upgrade rust nightly, we need to add a clippy allow rule then. Same for unsigned_abs.

The thing is that if there's no need for the types (and most of them will not need to be `pub`) the additions of `Eq` required will become minimal. 

I think is better to wait until indeed we need to do the change.

CPerezz commented:
>BTW any plan to upgrade to latest nightly and Rust2021?)

I would like us to move there. But latest nightly clippy has an ICE (so I prefer to wait). Also, as for Rust 2021 edition, there's other priorities than this one. And we don't need `resolver=2` which is the most interesting benefit.

So we might wait a bit or let some contributor to do the update.

ChihChengLiang reviewed (approved):
LGTM
